### PR TITLE
cmd: hardening threshold parameter checks

### PR DESCRIPTION
**Pull request by aadomn**
Ensure the threshold parameter given as input to the `create dkg` and `create cluster` commands is sound.

category: bug
ticket: #3241

### CODE REVIEW

**Comment by gsora** [comment]
@aadomn could you please add unit tests for those two edge cases?

**Comment by aadomn** [comment]
> @aadomn could you please add unit tests for those two edge cases?

@gsora done :heavy_check_mark:

**Review by gsora — APPROVED** [review]

File: cmd/createcluster.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ const (
 	zeroAddress    = "0x0000000000000000000000000000000000000000"
 	defaultNetwork = "mainnet"
 	minNodes       = 3
+	minThreshold   = 2
```
Comment (pinebit) [inline cmd/createcluster.go:54]: Shall this be `ceil(n*2/3)`?

File: cmd/createcluster.go [hunk 2]
```diff
@@ -366,6 +368,15 @@ func validateCreateConfig(ctx context.Context, conf clusterConfig) error {
 		}
 	}
 
+	// check for threshold parameter
+	if conf.Threshold < minThreshold {
+		return errors.New("threshold cannot be smaller than 2", z.Int("threshold", conf.Threshold))
```
Comment (pinebit) [inline cmd/createcluster.go:373]: Don't mention 2 in the message, use `z.Int` param similar to how `create dkg` is made.

**Review by pinebit — CHANGES_REQUESTED** [review]
Need changes

File: cmd/createcluster.go (hunk 2 above)
Comment (aadomn) [reply cmd/createcluster.go]: done

File: cmd/createcluster.go (hunk 1 above)
Comment (aadomn) [reply cmd/createcluster.go:55]: I left `minThreshold=2` here for tests and to clearly indicate that is the minimum threshold charon supports (when using 3 nodes only).
But indeed when validating the configuration it checks that the threshold is at least `ceil(n*2/3)`.
